nengo/config.py

```diff
@@ -84,6 +84,10 @@ def __repr__(self):
         return "<%s[%s]{%s}>" % (type(self).__name__,
                                  self._configures.__name__, ", ".join(params))
 
+    def __setstate__(self, state):
+        for k, v in state.items():
+            setattr(self, k, v)
```

Comment:
Just realized, it's not the `getattr(self._configures, key)` line in `get_param` that leads to the loop, but three lines above. The check `if key in self._extraparams` needs to look up the `_extraparams` attribute and does so via  `__getattr__` which again invokes `get_param`. That's the inifinite recursion!